### PR TITLE
[BugFix] Primary key vertical compaction maybe failed.

### PR DESCRIPTION
**Pull request by sevev**
## Why I'm doing:
During the vertical compaction of a PK table, if the input rowset contains only one non-overlapping rowset, we will create a `union_iterator` to read data. However, we will create `source_mask` to support vertical compaction but `union_iterator` does not support read data with `source_mask`, so the compaction task will fail and the output error is as follow:
```
W0618 20:10:32.126293 15920 rowset_merger.cpp:573] reader get next error. tablet=9720982, err=Not supported: get chunk with sources not supported
```

## What I'm doing:
Support `union_iterator` read data with `source_masks`.

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/union_iterator.cpp [hunk 1]
```diff
@@ -61,6 +61,7 @@ class UnionIterator final : public ChunkIterator {
     Status do_get_next(Chunk* chunk) override;
     Status do_get_next(Chunk* chunk, std::vector<uint32_t>* rowid) override;
     Status do_get_next(Chunk* chunk, std::vector<uint64_t>* rssid_rowids) override;
+    Status do_get_next(Chunk* chunk, std::vector<RowSourceMask>* source_masks) override { return do_get_next(chunk); }
```
Comment (luohaha) [inline be/src/storage/union_iterator.cpp:64]: No need to consider `source_masks` ?

Comment (sevev) [reply be/src/storage/union_iterator.cpp:64]: > No need to consider `source_masks` ?

I think it's unnecessary. The usage of `union_iterator` is to read data in the order of segments and it is no need to record the read order.

Comment (wyb) [reply be/src/storage/union_iterator.cpp:64]: add comments in `rowset_merger`?

Comment (sevev) [reply be/src/storage/union_iterator.cpp:66]: > add comments in `rowset_merger`?

done.

**Review by wyb — DISMISSED** [review]

**Review by decster — APPROVED** [review]

**Review by luohaha — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**